htlcswitch/switch.go

```diff
@@ -985,6 +985,35 @@ func (s *Switch) handlePacketForward(packet *htlcPacket) error {
 			return s.failAddPacket(packet, failure)
 		}
 
+		// Check if the incoming channel link has been Disabled by the user.
+		// If so, reject the HTLC.
+		s.indexMtx.RLock()
```

Comment:
Rather than check this at the link level, once we already have the packet, I think we may want to isntead do this within the switch, where we check to see if a link is even able to carry an HTLC: https://github.com/lightningnetwork/lnd/blob/master/htlcswitch/switch.go#L1047

Similarly, we want to ensure that we don't allow an HTLC to even reach the switch if it's sent over a disabled channel, so we'd want to add similar logic when we accept an inbound HTLC. 

I think the `rejectHTLC` logic above could also similarly be moved into these two areas to further consolidate the logic in this area of the codebase.